core/blocks.js

```diff
@@ -35,3 +35,12 @@ goog.provide('Blockly.Blocks');
  * @type {!Object<string,Object>}
  */
 Blockly.Blocks = new Object(null);
+
+/**
+ * Get the default checkbox state of a given block. To be overridden.
```

Comment:
I'm really leery of having the checkbox state be on the block instead of on the flyout.  For instance, what if you have the same block in multiple categories in the toolbox, with different IDs?  (We can't do much if they have the same ID.)  The state of the checkbox in the flyout isn't intrinsic to the block prototype, and not every block instance is in a flyout--not even every BlockSvg instance.

If this is just a hack for the sake of testing, please add a TODO with a linked github issue to every place where you're hacking for the sake of adding this feature, so we can find all of the places to unhack later.